src/tools/ScenarioMeasurement/Startup/ParserUtility.cs

```diff
@@ -0,0 +1,94 @@
+﻿using Microsoft.Diagnostics.Tracing;
+using System;
+using System.Collections.Generic;
+using System.CommandLine;
+using System.Data;
+using System.Data.Common;
+using System.Diagnostics;
+using System.Text.RegularExpressions;
+
+namespace ScenarioMeasurement
+{
+    class ParserUtility
+    {
+
+        public static bool MatchProcessStart(TraceEvent evt, TraceSourceManager source, string processName, IList<int> pids, string commandLine)
+        {
+            return MatchCommandLine(evt, source, commandLine) &&
+                   MatchProcessName(evt, source, processName) &&
+                   MatchProcessID(evt, source, pids);
+        }
+
+        public static bool MatchCommandLine(TraceEvent evt, TraceSourceManager source, string commandLine)
+        {
+            if (source.IsWindows)
+            {
+                if (((string)GetPayloadValue(evt, "CommandLine")).Trim() != commandLine)
+                {
+                    return false;
+                }
+            }
+            // Match the command line as well because pids might be reused during the session
+            return true;
```

Comment:
This seems to always return true on Linux. Is this intended?

Reply:
Yes because there's no commandline field in Linux kernel event payload.

Reply:
We can probably avoid getting a false positive by doing this platform-specific check in the parser classes, but I don't feel confident that 1) we'd have a bunch of if-else again in the parser classes or 2)encounter exception when we forget to have the conditionals. Maybe we can create another child class in ```ParserUtility.cs``` that  extends the functionality of boolean ---```Match:true```, ```NotMatch:false```, ```NotApplicable:true``` ?